make.paws/tests/testthat/test_sdk_helper.R

```diff
@@ -0,0 +1,306 @@
+test_that("check list paws packages", {
+  tmp_dir <- tempdir()
+  dir_list <- c("paws", "paws.cat1", "paws.cat2", "fake")
+  sapply(file.path(tmp_dir, dir_list), dir.create)
+
+  expect_equal(
+    basename(list_paws_pkgs(tmp_dir)),
+    c("paws", "paws.cat1", "paws.cat2")
+  )
+  fs::dir_delete(file.path(tmp_dir, dir_list))
+})
+
+test_that("check list paws packages restrict package", {
+  tmp_dir <- tempdir()
+  dir_list <- c("paws", "paws.cat1", "paws.cat2", "fake")
+  sapply(file.path(tmp_dir, dir_list), dir.create)
+
+  expect_equal(
+    basename(list_paws_pkgs(tmp_dir, "paws.cat1")),
+    "paws.cat1"
+  )
+  fs::dir_delete(file.path(tmp_dir, dir_list))
+})
+
+test_that("check list paws category packages", {
+  tmp_dir <- tempdir()
+  dir_list <- c("paws", "paws.cat1", "paws.cat2", "fake")
+  sapply(file.path(tmp_dir, dir_list), dir.create)
+
+  pkgs_list <- list_paws_pkgs(tmp_dir)
+
+  expect_equal(
+    basename(list_cat_pkgs(pkgs_list)),
+    c("paws.cat1", "paws.cat2")
+  )
+  fs::dir_delete(file.path(tmp_dir, dir_list))
+})
+
+test_that("check list paws sub category packages", {
+  tmp_dir <- tempdir()
+  dir_list <- c("paws", "paws.cat1", "paws.cat1.p1", "paws.cat1.p2", "fake")
+  sapply(file.path(tmp_dir, dir_list), dir.create)
+
+  pkgs_list <- list_paws_pkgs(tmp_dir)
+
+  expect_equal(
+    basename(list_sub_cat_pkgs(pkgs_list)),
+    c("paws.cat1.p1", "paws.cat1.p2")
+  )
+  fs::dir_delete(file.path(tmp_dir, dir_list))
+})
+
+test_that("check local check_pkgs", {
+  mock_devtools_check <- mock2(
+    list(errors = "foo", warnings = "bar"),
+    list(errors = "foo", warnings = "bar", notes = "cho"),
```

Comment:
What are these extra return values for? It looks like the test expects the mock to return only the first.

Reply:
Oh, I see, it drops notes by default.